### PR TITLE
ci: add shellcheck

### PR DESCRIPTION
**Pull request by zll600**
resolve: https://github.com/mosn/htnn/issues/132 
use gha https://github.com/reviewdog/action-shellcheck to check shell scripts.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -91,3 +91,12 @@ jobs:
...
         make lint-remain
+
+    - name: shellcheck
+      uses: reviewdog/action-shellcheck@v1
+      with:
+        github_token: ${{ secrets.github_token }}
+        reporter: github-pr-review
+        path: "."
+        pattern: "*.sh"
+        exclude: "./.git/*"
```
Comment (spacewander) [inline .github/workflows/lint.yml:102]: We don't have shell script under this directory.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -91,3 +91,12 @@ jobs:
     - name: lint remain
       run: |
         make lint-remain
+
+    - name: shellcheck
```
Comment (spacewander) [inline .github/workflows/lint.yml:95]: Please name it in the pattern `lint xxx` and put it before `lint remain`.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (zll600) [reply .github/workflows/lint.yml:102]: we should set `exclude` to the directories which we do want to check. 
Like we do not need to check `.git` directory here.
ref: https://github.com/reviewdog/action-shellcheck?tab=readme-ov-file#exclude

File: .github/workflows/lint.yml (hunk 2 above)
Comment (zll600) [reply .github/workflows/lint.yml]: done, PTAL

File: .github/workflows/lint.yml (hunk 1 above)
Comment (spacewander) [reply .github/workflows/lint.yml:98]: We already use `*.sh` to limit the file we will check. Unless there is `.git/xx.sh`, we don't need to add another filter.

Comment (zll600) [reply .github/workflows/lint.yml:98]: agree, done

**Review by spacewander — APPROVED** [review]

**Comment by spacewander** [comment]
Merged. Thanks!

**Pull request merged**